code/modules/antagonists/changeling/changeling.dm

```diff
@@ -117,24 +115,24 @@
 		remove_changeling_powers()
 	//Repurchase free powers.
 	for(var/path in all_powers)
-		var/obj/effect/proc_holder/changeling/S = new path()
+		var/datum/action/changeling/S = new path()
 		if(!S.dna_cost)
 			if(!has_sting(S))
 				purchasedpowers += S
 				S.on_purchase(owner.current,TRUE)
 
-/datum/antagonist/changeling/proc/has_sting(obj/effect/proc_holder/changeling/power)
-	for(var/obj/effect/proc_holder/changeling/P in purchasedpowers)
+/datum/antagonist/changeling/proc/has_sting(datum/action/changeling/power)
+	for(var/datum/action/changeling/P in purchasedpowers)
```

Comment:
typeless loop

Reply:
What does this mean

Reply:
basically:
```
for(var/power in purchasedpowers)
    var/datum/action/changeling/P = power
    ....
```

everything in this list is already this type, so you don't need to do a type check(which is what this kind of loop does)

Reply:
appreciate it

Reply:
Microoptimizations